vernac/declareDef.ml

```diff
@@ -43,8 +43,10 @@ module Hook = struct
 end
 
 (* Locality stuff *)
-let declare_definition ~name ~scope ~kind ?hook_data ?(should_suggest=false) udecl ce imps =
+let declare_definition ~name ~scope ~kind ?hook_data udecl ce imps =
   let fix_exn = Declare.Internal.get_fix_exn ce in
+  let should_suggest = ce.Declare.proof_entry_opaque &&
+                       Option.is_empty ce.Declare.proof_entry_secctx in
```

Comment:
This means once we have the opaque `Lemma := ` it will also suggest, which is wrong. I think deciding to suggest should stay in lemmas.

Reply:
IMO the whole "suggest" business still needs some work; after thinking a bit tho I'd suggest you consider we keep it in `DeclareDef` for now, here are 3 reasons:
- in order to work it is accessing the low-level `proof_entry` struct, which we do make fully abstract in upcoming PRs, thus the logic has to be moved to declare
- having it back in lemmas means that other users of `DeclareDef` with opaque don't get suggestions, for example obligations / mutual definitions. IMO this is wrong, as this suggestion should be uniform. You may be right that this could be triggering a few strange situations, but IMO the correct thing to do is to put this in the common path and fix the callers so they can all provide the hints.
- `Lemma :=` is still not fully determined so I'd rather wait until it arrives to take decisions

So I still prefer to keep it at this level, specially due to point 2, but if you still want me to move it back no problem.